tango/workspaces/local_workspace.py

```diff
@@ -251,6 +251,10 @@ def step_starting(self, step: Step) -> None:
                 f"file at {self._step_lock_file(step)}.",
             )
 
+        if step_info.state == StepState.FAILED:
+            # Refresh environment metadata since it might be out-of-date now.
+            step_info.refresh()
```

Comment:
Why only when it failed?

Reply:
I think this is the only situation where it could be out-of-sync. @schmmd discovered this edge case. It happens like this:

1. Run a step that fails or cancel it as it's running.
2. Change something in your code or requirements without changing the VERSION of the step (perfectly reasonable thing to do).
3. Run it again.

If the step succeeds the environment metadata in step info is now out-of-date.